pkg/querier/series.go

```diff
@@ -106,6 +106,8 @@ func (s *V3ioSeries) initSeriesIter() {
 				s.set.logger.ErrorWith("Error reading chunk buffer", "Lset", s.lset, "err", err)
 			} else {
 				newIterator.chunks = append(newIterator.chunks, chunk)
+				newIterator.chunksMax = append(newIterator.chunksMax,
```

Comment:
if both arrays are of the same size, consider using a struct and do append at once